Source/WebKit/UIProcess/RemoteLayerTree/RemoteLayerTreeDrawingAreaProxy.mm

```diff
@@ -188,6 +188,42 @@
         callback(value, key.protectedProcess());
 }
 
+RemoteLayerTreeDrawingAreaProxy::ProcessState& RemoteLayerTreeDrawingAreaProxy::processStateForIdentifier(WebCore::ProcessIdentifier identifier)
+{
+    RefPtr webProcess = WebProcessProxy::processForIdentifier(identifier);
+    RELEASE_ASSERT(webProcess);
+
+    for (auto pair : m_remotePageProcessState) {
```

Comment:
Ideally we wouldn't be iterating through a HashMap like this.  Maybe we should keep a parallel HashMap of process identifier to WeakPtr of the state or something.

Reply:
This is sort of an existing problem, with looking up the process state using an `IPC::Connection` for all incoming messages.

It's a bit unfortunate that we go through the `RemotePageDrawingAreaProxy`'s `didReceiveMessage` function (which knows the process identifier and process proxy), through to `RemoteLayerTreeDrawingAreaProxy`'s receiver functions, which then have to reconstruct this info using the connection.

Maybe we should have the `RemotePageDrawingAreaProxy` do the lookup and store it (as the 'active' state or something). Sort of sad statefulness to have, but should work better.